core/src/main/java/org/apache/carbondata/core/datamap/Segment.java

```diff
@@ -64,6 +68,14 @@
 
   private String segmentString;
 
+  private String indexSize = "";
```

Comment:
Why is it a string instead of integer?

Reply:
Because loadMetadataDetails.getIndexSize returns String and i dont want to do any numeric computation

Reply:
It is better to consider as long than String to save memory

Reply:
done